$OpenBSD$
- 'clone' wants to build internal versions or dependancies.
  use our packages instead
- 'prepare' is useful only if we want to attempt building
  using ocp-build (which we don\'t do yet); the lang/ocaml exports
  OCAMLFIND_DESTDIR, this would hurt here
- 'with-ocamlbuild' doesn\'t build opam-check (needed for tests)
--- Makefile.orig	Wed Mar 13 09:36:37 2013
+++ Makefile	Mon Mar 18 13:32:19 2013
@@ -1,14 +1,13 @@
--include Makefile.config
+include Makefile.config
 
 LOCAL_OCPBUILD=./ocp-build/ocp-build
 OCPBUILD ?= $(LOCAL_OCPBUILD)
 SRC_EXT=src_ext
-TARGETS = opam opam-mk-repo
+TARGETS = opam opam-check opam-mk-repo
 
 .PHONY: all
 
-all: $(LOCAL_OCPBUILD) META
-	$(MAKE) clone
+all: prepare $(LOCAL_OCPBUILD) META
 	$(MAKE) compile
 
 scan: $(LOCAL_OCPBUILD)
@@ -34,15 +33,20 @@ with-ocamlbuild: autogen
 	  ocamlbuild $(OCAMLBUILD_FLAGS) opam-$$i.cma opam-$$i.cmxa &&\
 	  rm -f src/$$i/opam-$$i.mllib;\
 	done;\
-	ocamlbuild $(OCAMLBUILD_FLAGS) opamMain.native opam_mk_repo.native opam_repo_check.native &&\
-	ln -sf _build/src/client/opamMain.native opam &&\
-	ln -sf _build/src/scripts/opam_mk_repo.native opam-mk-repo &&\
-	ln -sf _build/src/scripts/opam_repo_check.native opam-repo-check
+	ocamlbuild $(OCAMLBUILD_FLAGS) \
+		opamMain.$(OCAMLBEST) \
+		opam_check.$(OCAMLBEST) \
+		opam_mk_repo.$(OCAMLBEST) \
+		opam_repo_check.$(OCAMLBEST) && \
+	ln -sf _build/src/client/opamMain.$(OCAMLBEST) opam &&\
+	ln -sf _build/src/scripts/opam_check.$(OCAMLBEST) opam-check &&\
+	ln -sf _build/src/scripts/opam_mk_repo.$(OCAMLBEST) opam-mk-repo &&\
+	ln -sf _build/src/scripts/opam_repo_check.$(OCAMLBEST) opam-repo-check
 
 $(LOCAL_OCPBUILD): ocp-build/ocp-build.boot ocp-build/win32_c.c
 	$(MAKE) -C ocp-build
 
-OCAMLFIND_DIR=$(shell ocamlfind printconf destdir)
+OCAMLFIND_DIR=$(shell unset OCAMLFIND_DESTDIR && ocamlfind printconf destdir)
 prepare: depends.ocp.in
 	sed "s|%{lib}%|$(OCAMLFIND_DIR)|g" depends.ocp.in > depends.ocp
 
